### PR TITLE
fix #814

### LINKED ISSUE #814: Any way to send image from HTTP (not HTTPS)?
Hello! Sorry for a newbie question, but I have been stuck with it for at least a few hours now. 

Is there any shortcut provided by Telegraf to be able to send pictures from HTTP remote host? In particular, I would like to embed them into **sendMediaGroup** : )

### PR DESCRIPTION
**Pull request by septs**
# Description

Use `node-fetch` extension definition.
automate select HTTP(S) Agent

```ts
    // node-fetch extensions to the whatwg/fetch spec
    agent?: Agent | ((parsedUrl: URL) => Agent);
```

see  https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6a29ba12c44624c16d68da6acfb267cdfcdbd700/types/node-fetch/index.d.ts#L34-L35

Fixes #814

## Type of change

Please delete options that are not relevant.

- [ ] Documentation (typos, code examples or any documentation update)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Node.js Version:
* Operating System:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/core/network/client.ts [hunk 1]
```diff
@@ -48,13 +48,18 @@ const DEFAULT_EXTENSIONS = {
...
-    keepAliveMsecs: 10000,
-  }),
+  agent(parsedURL) {
+    if (parsedURL.protocol !== 'https:') {
+      return http.globalAgent
+    }
+    return new https.Agent({
+      keepAlive: true,
+      keepAliveMsecs: 10000,
+    })
```
Comment (wojpawlik) [inline src/core/network/client.ts:61]: I don't think it should create new `Agent` on every request...

**Review by wojpawlik — CHANGES_REQUESTED** [review]
Is `agent` required? Why?

**Comment by septs** [comment]
> Is `agent` required? Why?

i not known

see https://github.com/telegraf/telegraf/commit/8fe5123e77383e04bba2dc6e2aff1f5ced613889
see https://github.com/telegraf/telegraf/commit/b96c59c4c12b1c02d5143a75d6d401c18130bc1c

**Comment by KnorpelSenf** [comment]
@sept Why closing? It's this fix not good?

**Comment by septs** [comment]
https://t.me/TelegrafJSChat/12673

CC @KnorpelSenf

**Comment by KnorpelSenf** [comment]
Thanks!

It's an issue if we split the communication between two platforms 😄 
Imagine how little people understand that aren't in the group chat … maybe we should provide those every time we take action based on discussions in the chat? @septs @wojpawlik

**Comment by wojpawlik** [comment]
This fix is overly complex, I'll try something simpler first, it's my next priority

**Pull request closed without merging**